### PR TITLE
MAT-6743 admin api to grant access

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-6743](https://jira.cms.gov/browse/MAT-6743)
(Optional) Related Tickets:

### Summary
admin api to grant access

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included and Code coverage is verified.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java [hunk 1]
```diff
@@ -33,29 +35,27 @@ public void createLibrarySet(
...
-  //    }
-  //  }
+  public LibrarySet updateLibrarySetAcls(String librarySetId, AclSpecification aclSpec) {
+    Optional<LibrarySet> optionalLibrarySet = librarySetRepository.findByLibrarySetId(librarySetId);
+    if (optionalLibrarySet.isPresent()) {
+      LibrarySet librarySet = optionalLibrarySet.get();
+      if (CollectionUtils.isEmpty(librarySet.getAcls())) {
+        librarySet.setAcls(List.of(aclSpec));
+      } else {
+        librarySet.getAcls().add(aclSpec);
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java]: we could check if ACL already exists. we might have made the same mistake on the measure side as well. I've noticed multiple entries for the same ACL.

**Comment by adongare** [comment]
Snyk is complaining about the XSS attack. you need to set the response content type.

File: src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java [hunk 2]
```diff
@@ -190,6 +192,27 @@ public ResponseEntity<String> changeOwnership(
     return response;
   }
 
+  @PutMapping("/{id}/grant")
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java]: Snyk is complaining about the XSS attack. you need to set the content type. please refer line 173

Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java]: thanks Ashok, I just fixed it based on your suggestion

File: src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java]: thanks Ashok, I have fixed it

File: src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java [hunk 3]
```diff
@@ -33,29 +36,49 @@ public void createLibrarySet(
...
+      String error =
+          String.format(
+              "Library with set id `%s` can not be shared. Library set may not exists.",
+              librarySetId, userId);
+      log.error(error);
+      throw new ResourceNotFoundException("LibrarySet", "id", librarySetId);
+    }
+  }
+
+  public AclSpecification addAcl(String userId, RoleEnum role) {
```
Comment (adongare) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java]: this not adding acl. maybe we could rename it to `createAcl`

Comment (adongare) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java]: also it could be private

Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/LibrarySetService.java]: thank you, I'll change it

**Review by adongare — APPROVED** [review]
